server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java

```diff
@@ -230,11 +226,29 @@ public TenantUserDto getTenant(String userId) throws KaaAdminServiceException {
         }
     }
 
+    private void checkTenantName(TenantUserDto tenantUser) throws KaaAdminServiceException {
+        String name = tenantUser.getTenantName();
+        if (isEmpty(name)) {
+            throw new IllegalArgumentException("Parameter tenantName is empty.");
+        }
+        List<TenantUserDto> tenants = new ArrayList<>();
+        tenants.addAll(getTenants());
```

Comment:
This is redundant. You should do this chek inside saveUser, or add dedicated query for this. We can't fetch all tenants each time we need to do validation